server: deflake TestAddNewStoresToExistingNodes

tbg opened the pull request:
This test originally took 40+s on my laptop because it sets up a nine-node
cluster using real disks.

This commit refactors it to use nine (sticky) in-memory stores instead. It now
runs ~4x as fast (on my machine).

Also, preemtively skip TestAddNewStoresToExistingNodes under race and deadlock.

Fixes https://github.com/cockroachdb/cockroach/issues/106355.

Epic: None
Release note: None

tbg committed:
server: speed up TestAddNewStoresToExistingNodes

This test originally took 40+s on my laptop because it sets up a nine-node
cluster using real disks.

This commit refactors it to use nine (sticky) in-memory stores instead. It now
runs ~4x as fast (on my machine).

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/106402)
<!-- Reviewable:end -->

tbg committed:
kvserver: skip TestAddNewStoresToExistingNodes under race, deadlock

It's a pretty heavyweight test. That sort of thing is bound to flake every now
and then - not useful.

RaduBerinde reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/106402#-:-N_6IRCN9VMafeky-11K:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @bananabrick and @erikgrinaker)



<!-- Sent from Reviewable.io -->

tbg commented:
TFTR!

bors r=raduberinde
